sci-physics/reduze/files/reduze-2.0.9-gcc6.patch

```diff
@@ -0,0 +1,58 @@
+--- a/reduze/amplitude.cpp
++++ b/reduze/amplitude.cpp
+@@ -782,7 +782,7 @@
+ 
+ 	ex resultLeadingNc = result;
+ 	for (exset::const_iterator it = found.begin(); it != found.end(); ++it) {
+-		VERIFY(is_a<function>(*it) && ex_to<function>(*it).get_name() == "Color");
++		VERIFY(is_a<GiNaC::function>(*it) && ex_to<GiNaC::function>(*it).get_name() == "Color");
+ 		VERIFY(is_a<lst>(it->op(0)));
+ 		ex res(1);
+ 		for (size_t i = 0; i < it->op(0).nops(); ++i)
+@@ -823,7 +823,7 @@
+ 	timer.restart();
+ 	exmap m;
+ 	for (exset::const_iterator it = found.begin(); it != found.end(); ++it) {
+-		VERIFY(is_a<function>(*it) && ex_to<function>(*it).get_name() == "Lorentz");
++		VERIFY(is_a<GiNaC::function>(*it) && ex_to<GiNaC::function>(*it).get_name() == "Lorentz");
+ 		VERIFY(is_a<lst>(it->op(0)));
+ 
+ 		ex res(1);
+--- a/reduze/files.cpp
++++ b/reduze/files.cpp
+@@ -420,7 +420,7 @@
+ 			get_filename_sectormappings(m.integralfamily()->name());
+ 	ofstream file(file_name.c_str());
+ 	if (!file)
+-		ABORT("Can't write to " << file);
++		ABORT("Can't write to " << file_name);
+ 	os << YAML::BeginMap << YAML::Key << "sectormappings" << YAML::Value;
+ 	os << m << YAML::EndMap;
+ 	file << os.c_str();
+@@ -617,7 +617,7 @@
+ 
+ bool is_readable_file(const string& filename) {
+ 	ifstream is(filename.c_str());
+-	bool ok = is;
++	bool ok = !is.fail();
+ 	is.close();
+ 	return ok;
+ }
+--- a/reduze/sectormappings.cpp
++++ b/reduze/sectormappings.cpp
+@@ -757,13 +757,13 @@
+ 	const Node& z_node = node["zero_sectors"];
+ 	for (Iterator n = z_node.begin(); n != z_node.end(); ++n)
+ 		for (Iterator s = n.second().begin(); s != n.second().end(); ++s)
+-			m.zero_sectors_.insert(*s);
++			m.zero_sectors_.insert((const int) *s);
```

Comment:
This calls `std::set::insert()` on [YAML::Node::operator int() const](http://docs.ros.org/diamondback/api/yaml_cpp/html/classYAML_1_1Node.html#a58b068d29c66db26313b0128191d808b) but in C++11/14 it seems to cause an ambiguity between `insert( const value_type& value )`, `insert( value_type&& value )`, and `insert( std::initializer_list<value_type> ilist )`.  Explicitly casting it to `const int` works to overcome this and is dialect neutral.